### DISCUSSION TITLE
how to trace message in BlockingQueue?

### DISCUSSION THREAD

**Discussion by qiaoxingxing** [apache/skywalking#11363] [Q&A] [answered]
Here is my code flow:
- Controller: It receives data and puts it into a BlockingQueue.
- ConsumerService: A `ScheduledExecutorService` consumes the data from the BlockingQueue at a fixed rate.

I want to trace this message properly.

I've gone through the manual API [Use Native Tracer](https://skywalking.apache.org/docs/skywalking-java/v9.0.0/en/setup/service-agent/java-agent/application-toolkit-tracer),
but it seems like I can only achieve this by using Inject/Extract to pass context information. However, these methods are typically used between different processes.

How can I properly achieve this within the same process?

**Comment by wu-sheng** [comment, marked as answer]
`Tracer#Capture` could build a snapshot for you. That snapshot is for cross thread propagation, rather than `inject/extract` which is for cross process propagation.

**Reply by qiaoxingxing** [reply]
Thanks a lot for your help!
